arrow-array/src/array/struct_array.rs

```diff
@@ -146,7 +146,9 @@ impl StructArray {
 
             if !f.is_nullable() {
                 if let Some(a) = a.logical_nulls() {
-                    if !nulls.as_ref().map(|n| n.contains(&a)).unwrap_or_default() {
+                    if a.null_count() > 0
+                        && !nulls.as_ref().map(|n| n.contains(&a)).unwrap_or_default()
```

Comment:
It probably isn't measurable, but as written this will now always count nulls even if they are unmasked

I think it would be better if we only count the nulls right before we error to avoid unecessary work

```suggestion
                    if !nulls.as_ref().map(|n| n.contains(&a)).unwrap_or_default()
                        && a.null_count() > 0
```

Reply:
Yes, I will change this. Thanks for catching.

Reply:
I've fixed this